Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,408 @@
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with a 'night time' street map style.
+        let map = Map(basemapStyle: .arcGISStreetsNight)
+        
+        /// The utility network for this sample.
+        private let utilityNetwork = UtilityNetwork(url: .featureServiceURL)
+        
+        /// The service geodatabase used to create the feature layer.
+        private let serviceGeodatabase = ServiceGeodatabase(url: .featureServiceURL)
+        
+        /// The service geodatabase feature layers.
+        private var layers: [FeatureLayer] = []
+        
+        /// The last element that was added to the list of filter barriers.
+        ///
+        /// When an element contains more than one terminal, the user should be presented with the
+        /// option to select a terminal. Keeping a reference to the last added element provides ease
+        /// of access to save the user's choice.
+        @Published private(set) var lastAddedElement: UtilityElement?
+        
+        /// The current tracing related activity.
+        @Published private(set) var tracingActivity: TracingActivity?
+        
+        /// The base trace parameters.
+        @Published private(set) var traceParameters = UtilityTraceParameters(traceType: .isolation, startingLocations: [])
+        
+        /// The point geometry of the starting location.
+        @Published private(set) var startingLocationPoint: Point!
+        
+        /// The filter barrier categories.
+        @Published private(set) var filterBarrierCategories: [UtilityCategory] = []
+        
+        /// The selected filter barrier category.
+        @Published var selectedCategory: UtilityCategory?
+        
+        /// The previously selected filter barrier category.
+        @Published private(set) var previousCategory: UtilityCategory?
+        
+        /// A Boolean value indicating whether to include isolated features in the
+        /// trace results when used in conjunction with an isolation trace.
+        @Published var includesIsolatedFeatures = true
+        
+        /// A Boolean value indicating whether the trace is completed.
+        @Published private(set) var traceCompleted = false
+        
+        /// A Boolean value indicating whether the filter barrier categories have loaded.
+        @Published private(set) var categoriesLoaded = false
+        
+        /// A Boolean value indicating if tracing is enabled.
+        @Published private(set) var traceEnabled = false
+        
+        /// A Boolean value indicating if the reseting the trace is enabled.
+        @Published private(set) var resetEnabled = false
+        
+        /// A Boolean value indicating if the user has added filter barriers to the trace parameters.
+        @Published private(set) var hasFilterBarriers = false
+        
+        /// A Boolean value indicating whether the terminal selection menu is open.
+        @Published var terminalSelectorIsOpen = false
```

Comment:
With the current code structure, it is kind of hard to move this variable purely to either the view or the model. I'll think about this again once other comments are addressed.